bots/flakes-refresh

```diff
@@ -0,0 +1,166 @@
+#!/usr/bin/python3
+# -*- coding: utf-8 -*-
+
+# This file is part of Cockpit.
+#
+# Copyright (C) 2017 Red Hat, Inc.
+#
+# Cockpit is free software; you can redistribute it and/or modify it
+# under the terms of the GNU Lesser General Public License as published by
+# the Free Software Foundation; either version 2.1 of the License, or
+# (at your option) any later version.
+#
+# Cockpit is distributed in the hope that it will be useful, but
+# WITHOUT ANY WARRANTY; without even the implied warranty of
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the GNU
+# Lesser General Public License for more details.
+#
+# You should have received a copy of the GNU Lesser General Public License
+# along with Cockpit; If not, see <http://www.gnu.org/licenses/>.
+
+import argparse
+import sys
+import time
+import os
+import urllib
+import socket
+import json
+
+sys.dont_write_bytecode = True
+
+import task
+
+NUMBER_OPEN_ISSUES = 7            # How many issues do we want to have open at a given time?
+
+# How far back does our data go?  If a flake gets fixed but is still
+# flaky after this many days, the bots open another issue.
+
+WINDOW_DAYS = 90
+
+# This parses the output JSONL format discussed here, where various
+# values are grouped:
+#
+# https://github.com/cockpit-project/cockpituous/blob/master/learn/README.md
+
+# Here we're looking for a field in a record that only has one value
+def value(record, field):
+    values = record.get(field, [])
+    if len(values) == 1:
+        return values[0][0] or ""
+    return None
+
+# Here we're looking for the count of a specific field/value in the record
+def count(record, field, only):
+    values = record.get(field, [])
+    for value, count in values:
+        if value != only:
+            continue
+        return count
+    return 0
+
+def run(api, verbose=False, dry=False):
+    open_issues = api.issues(labels=[ "flake" ])
```

Comment:
That's the API query that will run in each iteration in each of the ~ 40 bots, right? My concern is that this will contribute to hitting the rate limiting root. But this is not a veto, we can start with this and move it to a weekly "issue lock" approach like npm-trigger. Or piggy-back on the existing learn-trigger/learn-tests.

Reply:
@mvollmer ... just add one more task to the bots/learn-trigger file here:

https://github.com/cockpit-project/cockpit/blob/master/bots/learn-trigger#L49

You can put "flake-refresh" there to address @martinpitt's concerns. This task will run on the same issues, but 4th in the list.

Reply:
That sounds good indeed - we can keep the code  separate, but re-use the trigger.

Reply:
Done.